src/main/java/com/google/sps/data/AvailabilityTimeSlots.java

```diff
@@ -0,0 +1,138 @@
+ * from datastore.
+ */
+public class AvailabilityTimeSlots {
+  private List<AvailabilityTimeSlot> timeSlots = new ArrayList<AvailabilityTimeSlot>();
+
+  /** Constructs an AvailabilityTimeSlots object by generating the timeSlots list. */
+  public AvailabilityTimeSlots(String timeZoneOffsetString) {
+    ZoneOffset timeZoneOffset = convertStringToOffset(timeZoneOffsetString);
+    ZoneId zoneId = ZoneId.ofOffset("UTC", timeZoneOffset);
+    ZonedDateTime today = Instant.now().atZone(zoneId);
+    String dayOfWeek = today.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.US);
+    int year = today.getYear();
+    int month = today.getMonthValue();
+    int dayOfMonth = today.getDayOfMonth();
+
+    // TODO: Generate a week of slots.
+    String date = generateDate(dayOfWeek, month, dayOfMonth);
+    List<String> times = generateTimes();
+    List<String> utcEncodings = generateUTCEncodings(year, month, dayOfMonth, zoneId);
+    List<Boolean> selectedStatuses = getSelectedStatuses();
+    timeSlots = generateTimeSlots(utcEncodings, times, date, selectedStatuses);
+  }
+
+  private ZoneOffset convertStringToOffset(String offsetString) {
+    int offsetTotalMinutes = Integer.parseInt(offsetString);
+    int offsetHours = offsetTotalMinutes / 60;
+    int offsetMinutes = offsetTotalMinutes % 60;
+    return ZoneOffset.ofHoursMinutes(offsetHours, offsetMinutes);
+  }
+
+  private String generateDate(String dayOfWeek, int month, int dayOfMonth) {
+    return dayOfWeek + " " + month + "/" + dayOfMonth;
+  }
+
+  private List<String> generateUTCEncodings(int year, int month, int dayOfMonth, ZoneId zoneId) {
+    List<String> utcEncodings = new ArrayList<String>();
+    int[] minutes = {0, 15, 30, 45};
+    for (int i = 8; i < 20; i++) {
+      for (int j = 0; j < 4; j++) {
+        LocalDateTime localTimeSlot = LocalDateTime.of(year, month, dayOfMonth, i, minutes[j]);
+        ZonedDateTime utcTimeSlot =
+            ZonedDateTime.of(localTimeSlot, zoneId).withZoneSameInstant(ZoneOffset.UTC);
+        String formattedUTCTimeSlot = utcTimeSlot.format(DateTimeFormatter.ISO_INSTANT);
+        utcEncodings.add(formattedUTCTimeSlot);
+      }
+    }
+    return utcEncodings;
+  }
+
+  // TODO: Access the time slots from data store to tell if they are selected or not.
+  private List<Boolean> getSelectedStatuses() {
+    List<Boolean> selectedStatuses = new ArrayList<Boolean>();
+    for (int i = 0; i < 48; i++) {
+      selectedStatuses.add(false);
+    }
+    return selectedStatuses;
+  }
+
+  private List<AvailabilityTimeSlot> generateTimeSlots(
+      List<String> utcEncodings, List<String> times, String date, List<Boolean> selectedStatuses) {
+    List<AvailabilityTimeSlot> availabilityTimeSlots = new ArrayList<AvailabilityTimeSlot>();
+    for (int i = 0; i < 48; i++) {
+      availabilityTimeSlots.add(
+          AvailabilityTimeSlot.create(
+              utcEncodings.get(i), times.get(i), date, selectedStatuses.get(i)));
+    }
```

Comment:
l don't know if this is a big deal but this code uses up twice the memory it actually needs. Could we generate these 48 slots/day in one really nested for loop? @acmcarther @jalexanderqed

Reply:
I think clarity would be lost in the process.